tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -396,6 +440,308 @@ def test_python_lambda_subscribe_sns_topic(
+            kinesis_name = f"test-kinesis-{short_uid()}"
+            kinesis_client.create_stream(StreamName=kinesis_name, ShardCount=1)
+
+            result = kinesis_client.describe_stream(StreamName=kinesis_name)["StreamDescription"]
+            kinesis_arn = result["StreamARN"]
+
+            # wait for stream-status "ACTIVE"
+            status = result["StreamStatus"]
+            if status != "ACTIVE":
+
+                def check_stream_active():
+                    state = kinesis_client.describe_stream(StreamName=kinesis_name)[
+                        "StreamDescription"
+                    ]["StreamStatus"]
+                    if state != "ACTIVE":
+                        raise Exception(f"StreamStatus is {state}")
+
+                retry(check_stream_active, retries=6, sleep=1.0, sleep_before=2.0)
+
+            queue_event_source_mapping = sqs_create_queue()
+            queue_failure_event_source_mapping_arn = sqs_queue_arn(queue_event_source_mapping)
+            destination_config = {
+                "OnFailure": {"Destination": queue_failure_event_source_mapping_arn}
+            }
+
+            result = lambda_client.create_event_source_mapping(
+                FunctionName=function_name,
+                BatchSize=1,
+                StartingPosition="TRIM_HORIZON",
+                EventSourceArn=kinesis_arn,
+                MaximumBatchingWindowInSeconds=1,
+                MaximumRetryAttempts=1,
+                DestinationConfig=destination_config,
+            )
+
+            event_source_mapping_uuid = result["UUID"]
+            event_source_mapping_state = result["State"]
+            if event_source_mapping_state != "Enabled":
+
+                def check_mapping_state():
+                    state = lambda_client.get_event_source_mapping(UUID=event_source_mapping_uuid)[
+                        "State"
+                    ]
+                    if state != "Enabled":
+                        raise Exception(f"State is {state}")
+
+                retry(check_mapping_state, retries=6, sleep_before=2.0, sleep=1.0)
+
+            message = {
+                "input": "hello",
+                "value": "world",
+                lambda_integration.MSG_BODY_RAISE_ERROR_FLAG: 1,
+            }
+            kinesis_client.put_record(
+                StreamName=kinesis_name, Data=to_bytes(json.dumps(message)), PartitionKey="custom"
+            )
+
+            def verify_failure_received():
+                result = sqs_client.receive_message(QueueUrl=queue_event_source_mapping)
+                msg = result["Messages"][0]
+                body = json.loads(msg["Body"])
+                assert body["requestContext"]["condition"] == "RetryAttemptsExhausted"
+                assert body["KinesisBatchInfo"]["batchSize"] == 1
+                assert body["KinesisBatchInfo"]["streamArn"] == kinesis_arn
+
+            retry(verify_failure_received, retries=50, sleep=5, sleep_before=5)
```

Comment:
wow, that's a lot of retries! Does it typically take this long to verify?

Reply:
not on LocalStack, but it can take some time on AWS